src/ctapipe/tools/compute_irf.py

```diff
@@ -0,0 +1,637 @@
+        default_value="BackgroundRate2dMaker",
+        help="The parameterization of the background rate to be used.",
+    ).tag(config=True)
+
+    energy_bias_resolution_maker_name = traits.ComponentName(
+        EnergyBiasResolutionMakerBase,
+        default_value="EnergyBiasResolution2dMaker",
+        help=(
+            "The parameterization of the bias and resolution benchmark "
+            "for the energy prediction."
+        ),
+    ).tag(config=True)
+
+    angular_resolution_maker_name = traits.ComponentName(
+        AngularResolutionMakerBase,
+        default_value="AngularResolution2dMaker",
+        help="The parameterization of the angular resolution benchmark.",
+    ).tag(config=True)
+
+    sensitivity_maker_name = traits.ComponentName(
+        SensitivityMakerBase,
+        default_value="Sensitivity2dMaker",
+        help="The parameterization of the point source sensitivity benchmark.",
+    ).tag(config=True)
+
+    spatial_selection_applied = Bool(
+        False,
+        help=(
+            "Compute an IRF after applying a direction cut (``SpatialSelection=RAD_MAX``) "
+        ),
+    ).tag(config=True)
+
+    aliases = {
+        "cuts": "IrfTool.cuts_file",
+        "gamma-file": "IrfTool.gamma_file",
+        "proton-file": "IrfTool.proton_file",
+        "electron-file": "IrfTool.electron_file",
+        "output": "IrfTool.output_path",
+        "benchmark-output": "IrfTool.benchmarks_output_path",
+        "chunk_size": "IrfTool.chunk_size",
+    }
+
+    flags = {
+        **flag(
+            "do-background",
+            "IrfTool.do_background",
+            "Compute background rate.",
+            "Do not compute background rate.",
+        ),
+        **flag(
+            "spatial-selection-applied",
+            "IrfTool.spatial_selection_applied",
+            "Compute an IRF after applying a direction cut (``SpatialSelection=RAD_MAX``).",
+            "Compute an IRF without any direction cut (``SpatialSelection=None``).",
+        ),
+    }
+
+    classes = (
+        [
+            EventLoader,
+        ]
+        + classes_with_traits(BackgroundRateMakerBase)
+        + classes_with_traits(EffectiveAreaMakerBase)
+        + classes_with_traits(EnergyDispersionMakerBase)
+        + classes_with_traits(PsfMakerBase)
+        + classes_with_traits(AngularResolutionMakerBase)
+        + classes_with_traits(EnergyBiasResolutionMakerBase)
+        + classes_with_traits(SensitivityMakerBase)
+    )
+
+    def setup(self):
+        """
+        Initialize components from config and load g/h (and theta) cuts.
+        """
+        self.opt_result = OptimizationResult.read(self.cuts_file)
+        if (
+            self.spatial_selection_applied
+            and self.opt_result.spatial_selection_table is None
+        ):
+            raise ToolConfigurationError(
+                f"{self.cuts_file} does not contain any direction cut, "
+                "but --spatial-selection-applied was given."
```

Comment:
I am really not sure if it is worth changing the terminology here... we will constantly tell people "We call point-like IRFs now "spatial-selection-applied IRFs". And it won't ever catch on because it is really unweildy compared to just saying "point-like".

cc @kosack